src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.stream.IntStream;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements = new Object[DEFAULT_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        if (elements.length == size) {
+            Object[] newElements = new Object[elements.length * 3 / 2 + 1];
+            System.arraycopy(elements, 0, newElements, 0, size);
+            elements = newElements;
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("index is false");
+        }
+        Objects.checkIndex(index, size + 1);
```

Comment:
see common mistakes to fix it

Reply:
![image](https://user-images.githubusercontent.com/80981426/213876272-531475e3-e728-4079-9814-7d08b7500ee8.png)